File: backend/deliberation-session-rattrapage-service/src/main/java/fsm/miaad/models/PlanDeValidation.java
package fsm.miaad.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data @AllArgsConstructor @NoArgsConstructor @ToString @Builder
public class PlanDeValidation {
    private String CNE;
    private Long moduleID;
    private String semestreID;
    private Long filiereID;
    private Long elementID;

    private Float noteTP;
    private Float noteExam;
    private Float noteFinale;
    private Resultat resultat;
}

File: frontend/systeme-interne/src/app/models/PlanDeValidationModel/plandevalidation.ts
import { Plandevalidationid } from "./plandevalidationid";

export class Plandevalidation {
  id!: Plandevalidationid;
  cne!: string;
  moduleID!: number;
  moduleIntitule?:string;
  semestreID!: string;
  filiereID!: number;
  filiereNom?:string
  elementID!: number;
  partieCours?:string;
  noteTP!: number;
  noteExam!: number;
  noteFinale?: number;
  resultat?: string; // Assuming Resultat is a string enum in your Spring Boot project
}
